smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -382,7 +382,8 @@ private void writeHeaders(
                                         String headerValue = getInputValue(context, binding.getLocation(),
                                                 memberLocation + "![suffix]", binding.getMember(), target);
                                         // Append the prefix to key.
-                                        writer.write("acc[$S + suffix] = $L;", binding.getLocationName(), headerValue);
+                                        writer.write("acc[$S + suffix!.toLowerCase()] = $L;",
+                                                binding.getLocationName().toLowerCase(Locale.US), headerValue);
                                         writer.write("return acc;");
```

Comment:
nit: this section can be simplified to use destructuring as follows:
```java
writer.openBlock("(acc: any, suffix: string) => ({", "}), {}",
    () -> {
        // Use a ! since we already validated the input member is defined above.
        String headerValue = getInputValue(context, binding.getLocation(),
                memberLocation + "![suffix]", binding.getMember(), target);
        writer.write("...acc,");
        // Append the prefix to key.
        writer.write("[`$L$${suffix.toLowerCase()}`]: $L,",
                binding.getLocationName(), headerValue);
    });
```

TypeScript's [Definite Assignment Assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#definite-assignment-assertions) is not required for `suffix` as it's always defined.

Reply:
You are right. Will address this.